firewood/src/db.rs

```diff
@@ -272,6 +273,43 @@ pub struct DbRev<S> {
     merkle: Merkle<S>,
 }
 
+#[async_trait]
+impl<S: ShaleStore<Node> + Send + Sync> api::DbView for DbRev<S> {
+    async fn root_hash(&self) -> Result<api::HashKey, api::Error> {
+        self.merkle
+            .root_hash(self.header.kv_root)
+            .map(|h| *h)
+            .map_err(|e| api::Error::IO(std::io::Error::new(ErrorKind::Other, e)))
+    }
+
+    async fn val<K: api::KeyType>(&self, key: K) -> Result<Option<Vec<u8>>, api::Error> {
+        let obj_ref = self.merkle.get(key, self.header.kv_root);
+        match obj_ref {
+            Err(e) => Err(api::Error::IO(std::io::Error::new(ErrorKind::Other, e))),
+            Ok(obj) => match obj {
+                None => Ok(None),
+                Some(inner) => Ok(Some(inner.as_ref().to_owned())),
+            },
+        }
+    }
+
+    async fn single_key_proof<K: api::KeyType, N: AsRef<[u8]> + Send>(
+        &self,
+        _key: K,
+    ) -> Result<Option<Proof<N>>, api::Error> {
+        todo!()
+    }
```

Comment:
If these are `todo` maybe we can do it in a follow up with real implementations?

Reply:
That's why this is 1/3 :)